### PR TITLE
feat(pubsublite): assigning subscriber implementation

### PR DESCRIPTION
**Pull request by tmdiep**
`assigningSubscriber` uses the Pub/Sub Lite partition assignment service to listen to its assigned partition numbers and dynamically add/remove `singlePartitionSubscribers`.

### CODE REVIEW

**Comment by tmdiep** [comment]
WANT_LGTM=@dpcollins-google for checking assigner and assigningSubscriber logic.
Other reviewers optional. Feel free to leave comments.

File: pubsublite/internal/wire/assigner.go
Comment (dpcollins-google) [inline pubsublite/internal/wire/assigner.go:48]: nit. consider a typedef

File: pubsublite/internal/wire/assigner.go
Comment (dpcollins-google) [inline pubsublite/internal/wire/assigner.go:54]: How can this fail?

File: pubsublite/internal/wire/assigner.go
Comment (dpcollins-google) [inline pubsublite/internal/wire/assigner.go:119]: hmm. How did you configure the retryable stream to not have an initial response?

File: pubsublite/internal/wire/subscriber.go
Comment (dpcollins-google) [inline pubsublite/internal/wire/subscriber.go:500]: I would actually delete this class entirely. assigningSubscriber is a generalization of this, where the "assigner" only ever produces a single set of values.

File: pubsublite/internal/wire/assigner.go
Comment (dpcollins-google) [reply pubsublite/internal/wire/assigner.go:54]: Oh wow the uuid library can fail XD weird....

**Review by dpcollins-google — CHANGES_REQUESTED** [review]
LGTM after changes from a pub/sub lite perspective, I'd prefer if you get someone familiar with go to review as well.

File: pubsublite/internal/wire/subscriber.go
Comment (tmdiep) [reply pubsublite/internal/wire/subscriber.go:492]: True.  Although we can avoid extra complexity in the `assigner` to handle this mode, avoid creating the PartitionAssignmentClient, etc. So I might leave `newMultiPartitionSubscriber`, as it's only a few lines of extra code (+tests) anyway.

File: pubsublite/internal/wire/assigner.go
Comment (tmdiep) [reply pubsublite/internal/wire/assigner.go:48]: Ah yeah... I used raw types in the API due to other reviewers' comments and just kept it consistent in the wire package. Will think about defining strong types (in a separate PR, since it requires unrelated changes). Since this is called `partitionSet`, it might be ok.

File: pubsublite/internal/wire/assigner.go
Comment (tmdiep) [reply pubsublite/internal/wire/assigner.go:119]: The second return value of `initialRequest` is a bool indicating whether an initial response is expected. I will refactor it to a strong type in a separate PR (pulls in other unrelated changes) to make it clearer.

As for why - in long running integration tests, I found that when the partition assignment stream reconnects and the client UUID hasn't changed, the server doesn't bother to return the initial set of partition assignments (so my test would just hang waiting for it, then timeout and fail).

**Comment by tmdiep** [comment]
WANT_LGTM=@hongalex for Go, and also Dan doesn't have write access and can't approve anyway.

**Review by manuelmenzella-google — APPROVED** [review]

**Pull request merged**
